### PR TITLE
DOCS-3091: Add polymorphism support to POST /orders

### PR DESCRIPTION
**Pull request by Kris-MultiSafepay**
Do the changes meet MultiSafepay Docs' Definition of Done?
- Style guide, tone of voice, and naming conventions
- Tested
- Reviewed on Stoplight, Redocly
- Notified stakeholders
- Pass GitHub Actions

### CODE REVIEW

File: openapi/components/schemas/orders/CreditCard.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      For more information, see [Direct vs redirect API requests](https://docs.multisafepay.com/developer/direct-vs-redirect).
+      
+      **Note**: To process `direct` credit card payments, you must have PCI DSS certification, see [Handling cardholder data](https://docs.multisafepay.com/features/handling-cardholder-data/).
+    type: string
+    enum: [direct, redirect]
+    example: redirect
+  gateway:
+    allOf:
+      - description: |
+          The unique gateway identifier for the payment method.
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/CreditCard.yaml:18]:
```suggestion
          The unique identifier for the payment method (gateway).
```

File: openapi/components/schemas/orders/CreditCard.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+          If `type` is set to redirect and `gateway` is set to `CREDITCARD`, a payment page is returned that automatically detects the customer's credit card scheme based on the cardholder data.
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/CreditCard.yaml:20]:
```suggestion
         For `redirect` requests using the generic `CREDITCARD` gateway, a payment page is returned that automatically detects the card scheme.
```

File: openapi/components/schemas/orders/FastCheckout.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
+title: FastCheckout order
+description: FastCheckout order
+properties:
+  type:
+    description: |
+      Specifies the order type.
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/FastCheckout.yaml:6]:
```suggestion
    description: Specifies the order type.  
```

File: openapi/components/schemas/orders/FastCheckout.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+properties:
+  type:
+    description: |
+      Specifies the order type.  
+    type: string
+    enum: [checkout]
+    example: checkout
+  gateway:
+    allOf:
+      - description: "The unique gateway identifier for the payment method.<br><br>If unset, a FastCheckout page with all available payment methods is returned."
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/FastCheckout.yaml:12]:
```suggestion
      - description: "The unique identifier for the payment method (gateway).<br><br>If unset, a FastCheckout page with all available payment methods is returned."
```

File: openapi/components/schemas/orders/FastCheckout.yaml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      notification_method:
+        $ref: '../NotificationMethod.yaml'
+      redirect_url:
+        $ref: '../RedirectUrl.yaml'
+      cancel_url:
+        $ref: '../CancelUrl.yaml'
+      close_window:
+        $ref: '../CloseWindow.yaml'
+      settings:
+
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/FastCheckout.yaml:45]: Is something missing here?

File: openapi/components/schemas/orders/PayLater.yaml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      - `direct`: Redirects the customer to an authentication or success page, depending on the payment method. 
+      - `paymentlink`: Creates a [payment link](https://docs.multisafepay.com/payment-links/).
+      
+      For more information, see [Direct vs redirect API requests](https://docs.multisafepay.com/developer/direct-vs-redirect).
+    type: string
+    enum: [direct, redirect]
+    example: redirect
+  gateway:
+    allOf:
+      - description: "The unique gateway identifier for the payment method.<br><br>If unset, and `type` is set to `redirect`, a MultiSafepay payment page with all available payment methods is returned."
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/PayLater.yaml:16]:
```suggestion
      - description: "The unique identifier for the payment method (gateway).<br><br>If unset, and `type` is set to `redirect`, a MultiSafepay payment page with all available payment methods is returned."
```

File: openapi/components/schemas/orders/PaymentPage.yaml [hunk 7]
```diff
@@ -0,0 +1,78 @@
+title: Payment page
+description: Payment page
+properties:
+  type:
+    description: |
+      - `redirect`: Redirects the customer to a [MultiSafepay payment page](https://docs.multisafepay.com/payment-pages/), where they can select a payment method and complete payment. 
+      - `paymentlink`: Creates a [payment link](https://docs.multisafepay.com/payment-links/), which is useful when manually managing the order through the merchant dashboard.
+      
+    type: string
+    enum: [redirect, paymentlink]
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/PaymentPage.yaml:10]:
```suggestion
    enum: [paymentlink, redirect]
```

File: openapi/components/schemas/orders/PrepaidCard.yaml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+      - `paymentlink`: Creates a [payment link](https://docs.multisafepay.com/payment-links/).
+      
+      For more information, see [Direct vs redirect API requests](https://docs.multisafepay.com/developer/direct-vs-redirect).
+    type: string
+    enum: [direct, redirect]
+    example: redirect
+  gateway:
+    allOf:
+      - description: |
+          The unique gateway identifier for the payment method.
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/PrepaidCard.yaml:17]:
```suggestion
          The unique identifier for the payment method (gateway).
```

File: openapi/components/schemas/orders/Wallet.yaml [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+      - `redirect`: Redirects the customer to a [MultiSafepay payment page](https://docs.multisafepay.com/payment-pages/), where they can select a payment method and complete payment. 
+      - `direct`: Redirects the customer to an authentication or success page, depending on the payment method. 
+      
+      For more information, see [Direct vs redirect API requests](https://docs.multisafepay.com/developer/direct-vs-redirect).
+    type: string
+    enum: [direct, redirect]
+    example: redirect
+  gateway:
+    allOf:
+      - description: "The unique gateway identifier for the payment method.<br><br>If unset, and `type` is set to `redirect`, a MultiSafepay payment page with all available payment methods is returned."
```
Comment (FrancescaNCook) [inline openapi/components/schemas/orders/Wallet.yaml:15]:
```suggestion
      - description: "The unique identifier for the payment method (gateway).<br><br>If unset, and `type` is set to `redirect`, a MultiSafepay payment page with all available payment methods is returned."
```

**Review by FrancescaNCook — APPROVED** [review]

File: openapi/components/schemas/GatewayInfo.yaml [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+      - Apple Developer – [ApplePayPaymentToken](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaypaymenttoken)
+
+      For Google Pay direct orders: The payment token returned from the client-side Google Pay API call.
+
+
+      Access the token at `PaymentData.PaymentMethodData.PaymentMethodTokenizationData.token`.
+
+
+      See [Google Pay direct integration](https://docs.multisafepay.com/payments/methods/wallet/googlepay/direct-integration/).
+    type: string
```
[the remark this answers is not in the document]
Comment (Kris-MultiSafepay) [reply openapi/components/schemas/GatewayInfo.yaml:72]: We could, but they're very long encrypted strings, like: "b94c896d0bcea31196f6ddedad3aa57ae3ec45cfe54f633da5a1b73425931649b5d2cbd0f17e590234c9f8325ddffa3cf9c469df5f018cdb9474f9fdbcb031e4b94c896d0bcea31196f6ddedad3aa57ae3ec45cfe54f633da5a1b73425931649b5d2cbd0f17e590234c9f8325ddffa3cf9c469df5f018cdb9474f9fdbcb031e4b94c896d0bcea31196f6ddedad3aa57ae3ec45cfe54f633da5a1b73425931649b5d2cbd0f17e590234c9f8325ddffa3cf9c469df5f018cdb9474f9fdbcb031e4b94c896d0bcea31196f6ddedad3aa57ae3ec45cfe54f633da5a1b73425931649b5d2cbd0f17e590234c9f8325ddffa3cf9c469df5f018cdb9474f9fdbcb031e4" (This is nonsense btw).

I think the default "string" is alright for now. Maybe we can consider truncating a token, but that might confuse users into thinking the token shouldn't be all that long.

File: openapi/components/schemas/Plugin.yaml [hunk 11]
```diff
@@ -0,0 +1,21 @@
+description: Specifies information about your integration, which is useful for debugging.
+properties:
+  shop:
+    description: The ecommerce platform that you use.
```
Comment (Kris-MultiSafepay) [inline openapi/components/schemas/Plugin.yaml:4]:
```suggestion
    description: The name of the application/ecommerce.
```

Comment (FrancescaNCook) [reply openapi/components/schemas/Plugin.yaml]: Ecommerce as a noun refers to the whole industry. What do you mean when you use it as a noun here? A webshop? An ecommerce platform?

Comment (Kris-MultiSafepay) [reply openapi/components/schemas/Plugin.yaml]: I've changed it to 'application'. Basically:

- `shop` is used to identify the application/website of the merchant
- `shop_version` is used to identify the version of the application/website
- `plugin_version` is used to identify the version of the integration of MultiSafepay into the application/website.

I think application will make sense, but it gets confusing when the integration is an application e.g. with Lightspeed. Lightspeed is an application and the integration of MultiSafepay into Lightspeed is through an in-app app.

Comment (FrancescaNCook) [reply openapi/components/schemas/Plugin.yaml]: Could we say "ecommerce application"? I wonder that the context isn't quite clear enough for a very broad term like "application"?

**Pull request merged**